code/game/objects/machinery/vending/quick_vendor.dm

```diff
@@ -0,0 +1,120 @@
+GLOBAL_LIST_INIT(quick_loadouts, init_quick_loadouts())
+
+///The list is shared across all quick vendors, but they will only display the tabs specified by the vendor, and only show the loadouts with jobs that match the displayed tabs.
+/proc/init_quick_loadouts()
+	. = list()
+	var/list/loadout_list = list(
+		/datum/outfit/quick/tgmc/marine/standard_assaultrifle,
+		/datum/outfit/quick/tgmc/marine/standard_laserrifle,
+		/datum/outfit/quick/tgmc/marine/standard_machinegunner,
+		/datum/outfit/quick/tgmc/marine/medium_machinegunner,
+		/datum/outfit/quick/tgmc/marine/standard_shotgun,
+		/datum/outfit/quick/tgmc/marine/standard_lasercarbine,
+		/datum/outfit/quick/tgmc/engineer/rrengineer,
+		/datum/outfit/quick/tgmc/engineer/sentry,
+		/datum/outfit/quick/tgmc/engineer/demolition,
+		/datum/outfit/quick/tgmc/corpsman/standard_medic,
+		/datum/outfit/quick/tgmc/smartgunner/standard_sg,
+		/datum/outfit/quick/tgmc/smartgunner/minigun_sg,
+		/datum/outfit/quick/tgmc/leader/standard_assaultrifle,
+		/datum/outfit/quick/som/marine/mpi,
+		/datum/outfit/quick/som/marine/charger,
+		/datum/outfit/quick/som/marine/caliver,
+		/datum/outfit/quick/som/veteran/charger,
+		/datum/outfit/quick/som/veteran/culverin,
+	)
+
+	for(var/X in loadout_list)
+		.[X] = new X
+
+
+/obj/machinery/quick_vendor
+	name = "Kwik-E-Quip vendor"
+	desc = "An advanced vendor to instantly arm soldiers with specific sets of equipment, allowing for immediate combat deployment."
+	icon = 'icons/obj/machines/vending.dmi'
+	icon_state = "specialist"
+	density = TRUE
+	anchored = TRUE
+	layer = BELOW_OBJ_LAYER
+	req_access = null
+	req_one_access = null
+	interaction_flags = INTERACT_MACHINE_TGUI
+	///The faction of this quick load vendor
+	var/faction = FACTION_NEUTRAL
+
+/obj/machinery/quick_vendor/som
+	faction = FACTION_SOM
+
+/obj/machinery/quick_vendor/can_interact(mob/user)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(!ishuman(user))
+		return FALSE
+
+	var/mob/living/carbon/human/H = user
+	if(!allowed(H))
+		return FALSE
+
+	if(!isidcard(H.get_idcard())) //not wearing an ID
+		return FALSE
+
+	var/obj/item/card/id/I = H.get_idcard()
+	if(I.registered_name != H.real_name)
+		return FALSE
+	return TRUE
+
+/obj/machinery/quick_vendor/ui_interact(mob/living/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		if(faction == FACTION_SOM)
+			//SOM vendor has SOM specific job titles
+			ui = new(user, src, "Quickload_SOM")
+		else
+			ui = new(user, src, "Quickload_TGMC")
+		ui.open()
```

Comment:
early returns

Reply:
PS wouldnt a switch be better here for maintainability

Reply:
yep.